core/src/main.cpp

```diff
@@ -866,8 +881,16 @@ auto main(int _argc, char* _argv[]) -> int
 		// JSON configs needs to be in main scope to last the entire duration of the program
 		nlohmann::json oi_config;
 		nlohmann::json endpoint_config;
-		load_oidc_configuration(http_server_config, oi_config, endpoint_config);
 
+		// Check if OIDC config exists, skip setup if missing.
+		if (http_server_config.contains(json::json_pointer{"/authentication/openid_connect"})) {
+			if (!load_oidc_configuration(http_server_config, oi_config, endpoint_config)) {
```

Comment:
We can avoid multiple lookups by using `.find`.

If the iterator is valid, pass its value to `load_oidc_configuration`.

Reply:
Done.